mix.exs

```diff
@@ -14,6 +14,9 @@ defmodule Absinthe.Mixfile do
       start_permanent: Mix.env() == :prod,
       package: package(),
       source_url: @source_url,
+      preferred_cli_env: [
```

Comment:
This isn't sufficient to avoid compiling the entire app again in `dev` since it needs to compile to _run_ the `mix dialyzer` task itself

Reply:
No, running just mix itself does not lead to starting the compilation process. It firstly reads mix.exs file content, then decides that should be done (evaluating aliases, setting up proper environment and paths for the compiler), and only after that runs the desired action.

Because `dialyzer` step now have default `test` environment, mix will firstly resolve all configs and paths for this specific environment, then run `dialyzer` itself, which then run `compile` step internally. You can check CI job on my fork to see that there is no additional compilation step performed (https://github.com/dolfinus/absinthe/runs/2160694599?check_suite_focus=true, see `Run dialyzer` step).

Reply:
hmm, looks like that's correct - i tried it locally before but must have done something wrong...